### DISCUSSION TITLE
MacOS Resolution off

### DISCUSSION THREAD

**Discussion by ThijmenL98** [AvaloniaUI/Avalonia#7919] [Q&A]
My application is built to be 1900x1000 (approximately) in size. Yet, a friend of mine who is testing my application, who owns a Macbook M1 Air (resolution 2560x1600 pixels), is unable to make it any smaller, and the application looks almost full-screen on his device. 

Does anyone know what could cause this?

Someone pointed out that it might be something with the retina display numbers not coming through properly. And assumed removing min widths and heights for a Window, that it would work, however, this is yet untested.

**Comment by sn4k3** [comment]
Could be screen scalling? And yes macOS report strange resolutions since monjave, way less than real.

I do also use auto sizes, and those are a hell to mantain and need hacks (code behind) to mantain. The auto size in avalonia are kinda broken.

My samples:
```C#
if (OperatingSystem.IsMacOS()) // Fix macOS scaling information
        {
            var monjave = new Version(10, 14, 6);
            if (Environment.OSVersion.Version.CompareTo(monjave) >= 0)
            {
                General.WindowsTakeIntoAccountScreenScaling = false;
            }
        }
```

```C#
    public static Screen GetCurrentScreen(this Window window)
    {
        return //window.Screens.ScreenFromVisual(window) ??
            window.Screens.ScreenFromVisual(App.MainWindow) ??
            window.Screens.Primary ??
            window.Screens.All[0];
    }

    public static System.Drawing.Size GetScreenWorkingArea(this Window window)
    {
        var screen = window.GetCurrentScreen();
            
        return new System.Drawing.Size(
            UserSettings.Instance.General.WindowsTakeIntoAccountScreenScaling ? (int)(screen.WorkingArea.Width / screen.PixelDensity) : screen.WorkingArea.Width,
            UserSettings.Instance.General.WindowsTakeIntoAccountScreenScaling ? (int)(screen.WorkingArea.Height / screen.PixelDensity) : screen.WorkingArea.Height);
    } 
```

```C#
    public double WindowMaxWidth => this.GetScreenWorkingArea().Width - UserSettings.Instance.General.WindowsHorizontalMargin;

    public double WindowMaxHeight => this.GetScreenWorkingArea().Height - UserSettings.Instance.General.WindowsVerticalMargin;
```

This will not fix your issue, but may give you some ideia

**Reply by maxkatz6** [reply]
Your issue seems to be related only to Screen API, where macOS might return scaled resolution. From what I understand.

**Reply by sn4k3** [reply]
Yes the Screen API return crazy scales on macOS, like 200%, starting on Monjave, so i have to ignore that information and use reported Resolution without scale, it works fine that way

**Comment by maxkatz6** [comment]
Yes, it's because of scaling, and that's normal. You can have similar experience on Windows by changing scaling in the OS settings.
IIRC Macbook M1 Air have 200% scaling reported by the system on retina display. Though I can be wrong with exact number.
